[WIP] Bump axios to ^1.4.0

JeanMarcGoepfert opened the pull request:
<!-- structure the Title above as the first line of a
     https://conventionalcommits.org/ message. example: "feat(buttons):
     add a muted button component". the title informs the semantic
     version bump if this PR is merged. -->

## Description

<!-- a summary of the changes introduced by this PR. this description
     may populate the commit body and versioned changelog if the PR is
     merged. -->

<!-- === GH HISTORY FORMAT FENCE === --> <!--
### [`%h`]({{.url}}/commits/%H) %s%n%n%b%n
--> <!-- === GH HISTORY FORMAT FENCE === -->
<!-- === GH HISTORY FENCE === -->
### [`5b43c69`](https://github.com/zendesk/zcli/pull/197/commits/5b43c697d977223580b3b7e76d2911e6ee5a154e) Bump axios to ^1.4.0



<!-- === GH HISTORY FENCE === -->

## Detail

<!-- supporting details; screen shot, code, etc. -->

<!-- closes GITHUB_ISSUE -->

## Checklist

- [ ] :guardsman: includes new unit and functional tests

anushkafka reviewed (approved):
I didn't realise we have replaced node-fetch with axios over time, I wonder why we replaced a minimalistic, light weight library with a dependency heavy library that shines in a browser based environment, considering this is a CLI tool 🤔

JeanMarcGoepfert commented:
> I didn't realise we have replaced node-fetch with axios over time, I wonder why we replaced a minimalistic, light weight library with a dependency heavy library that shines in a browser based environment, considering this is a CLI tool 🤔

Looks like it changed in [this PR](https://github.com/zendesk/zcli/pull/121) but no clear reason given 🤷.

token-cjg commented:
If I recall correctly I believe that the node-fetch -> axios change was made for security reasons, there were vulnerabilities with node-fetch that axios didn't have at the time.
